runtime/doc/api.txt

```diff
@@ -112,7 +112,9 @@ Basic types ~
   Dictionary (msgpack: map)
   Object
 <
-  Note: empty Array is accepted as a valid argument for Dictionary parameter.
+  Note: 
+    - An empty Array is accepted as a valid argument for Dictionary parameter.
+    - There is no Function type, so API functions cannot return Lua functions.
```

Comment:
I am not sure this is fully correct because of the `LuaRef` type

Reply:
If I change `cannot` to `should not` (or `shouldn't` - don't really know if there is a consensus for stuff like this in neovim?), would that be better? Or would you want a more detailed explanation?

I mainly wanted to add something to the docs to clarify the problem I ran into in #26531.

Reply:
I'm not a maintainer so I don't want to tell you what to do, just providing some more info :)

Definitely should be added to the docs as it could be confusing for new users, I agree!

Honestly it could be good to add a note about API functions that take functions as arguments specifically as well, since that's the case that caused the original issue.

As for specific instructions on what would be accepted in core though, I am not the person to ask haha